src/NHibernate/Util/SerializableSystemType.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace NHibernate.Util
+{
+	[Serializable]
+	internal sealed class SerializableSystemType : ISerializable
```

Comment:
This needs to be implemented using [`IObjectReference`](https://msdn.microsoft.com/en-us/library/system.runtime.serialization.iobjectreference) instead.

Reply:
I do not find any example about `IObjectReference` for something else than a custom singleton and I do not really see how it can be used for a type we do not define (`System.Type` in this case).
If we try `SerializationInfo.AddValue("someTypeField", someTypeField, typeof(SerializableSystemType))`:

- This violate (`AddValue`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.serializationinfo.addvalue) documentation which states:

> This parameter must always be the type of the object itself or of one of its base classes.

- This will style serialize the type fields I think (if it works).

Then on deserialization (if it indeed works for serializing despite what the documentation writes), how do we get information on what was actually serialized? By defining fields matching what a `System.Type` serialize? If it is that, it seems quite brittle compared to current implementation.

I guess that in fact I do not understand at all how to use `IObjectReference` in this context.